src/libpriv/rpmostree-origin.c

```diff
@@ -407,6 +435,24 @@ rpmostree_origin_set_jigdo_version (RpmOstreeOrigin *origin,
   origin->cached_jigdo_version = g_strdup (version);
 }
 
+/* The jigdoRPM is highly special; it doesn't live in the rpmdb for example, as
+ * that would be fully circular. Yet, it's of critical importance to the whole
+ * system; we want to render it on the client. For now, what we do is stick the
+ * EVR+A in the origin. That's the only data we really care about.
+ *
+ * Perhaps down the line, what we really want to do is store the whole Header at
+ * least somewhere hooked off the deployment (or perhaps imported itself into
+ * the pkgcache)?
+ */
+void
+rpmostree_origin_set_jigdo_description (RpmOstreeOrigin *origin,
+                                        DnfPackage      *package)
+{
+  g_assert_cmpint (origin->refspec_type, ==, RPMOSTREE_REFSPEC_TYPE_ROJIG);
+  g_key_file_set_string (origin->kf, "origin", "jigdo-evr", dnf_package_get_evr (package));
```

Comment:
Hmm, this gets a little confusing given that we already have a `jigdo-version` as well in the origin. It's also more cruft to maintain (e.g. I think we're missing cleaning those up when rebasing back to ostree mode here, right?). It would also be nice to keep the origin as pure input only for the upgrader, if possible.

Overall, I think I'm OK with this for now, though yeah, given its importance, I think it would make sense to just plop the whole header in the deployment.

Reply:
OK yeah, I'm going to try that.  You OK with `/usr/share/rpm-ostree/${jigdopkg}.hdr`?

Reply:
Yup, that works!

Reply:
OK after getting 80% of the way to implementing this I remember now the reason why the origin file exists outside of the deployment; putting data inside the deployment root changes it.  Now obviously technically we *could* just drop it in there post-deployment but...having files in the tree that *aren't* part of the ostree commit feels like it's going to trigger weird issues.

Reply:
Hmm, fair enough. Let's just keep the NEVRA in an xattr of the deployment dir? (And later if we see we really want the header, we can import it as well? Then the xattr acts as "ref" for looking it up and avoiding GC.)

Reply:
Even adding an xattr on the root would e.g. show up in ostree-based diffs.

Reply:
The whole Header here for a jigdoRPM is `930068` bytes.  That's kind of too much to e.g. hack as base64 into the origin.  So...I'm really torn because obviously it'd be quite nice down the line to be able to provide all the data that `rpm -q` wants.  Eh.

I think my vote for now is to go with this PR as is mostly, except maybe rename the variables for clarity?  Something like `jigdo-imported-evr`?

And fix the issue you pointed about about pruning them when rebasing to `ostree://`.

Reply:
Sure, let's go with that for now. Another place we can put it is at e.g. `/ostree/deploy/$osname/deploy/$csum.$i.rojighdr`? But yeah, you'd still want an easy way to pick up the NEVRA without opening the `Header` each time.

Reply:
> /ostree/deploy/$osname/deploy/$csum.$i.rojighd

The problem with that is libostree wouldn't know to clean it up.

Reply:
>  Something like `jigdo-imported-evr`?

Can we also rename `jigdo-version` to `jigdo-override-version` to make it clear it's the same idea as `override-commit`?

Reply:
Done in another commit :arrow_down: